src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java

```diff
@@ -0,0 +1,222 @@
+    public static final String APPELLANT = "Appellant";
+
+    private HearingsPartiesMapping() {
+
+    }
+
+    public static List<PartyDetails> buildHearingPartiesDetails(HearingWrapper wrapper) {
+
+        Appeal appeal = wrapper.getUpdatedCaseData().getAppeal();
+        Appellant appellant = appeal.getAppellant();
+
+        List<PartyDetails> partiesDetails = new ArrayList<>(buildHearingPartiesPartyDetails(appellant, appeal.getRep(), appeal.getHearingOptions(), appeal.getHearingType(), appeal.getHearingSubtype()));
+
+        List<CcdValue<OtherParty>> otherParties = wrapper.getUpdatedCaseData().getOtherParties();
+
+        if (nonNull(otherParties)) {
+            for (CcdValue<OtherParty> ccdOtherParty : otherParties) {
+                OtherParty otherParty = ccdOtherParty.getValue();
+                partiesDetails.addAll(buildHearingPartiesPartyDetails(otherParty, otherParty.getRep(), otherParty.getHearingOptions(), null, otherParty.getHearingSubtype()));
+            }
+        }
+
+        return partiesDetails;
+    }
+
+    public static List<PartyDetails> buildHearingPartiesPartyDetails(Party party, Representative rep, HearingOptions hearingOptions, String hearingType, HearingSubtype hearingSubtype) {
+        List<PartyDetails> partyDetails = new ArrayList<>();
+        partyDetails.add(createHearingPartyDetails(party, hearingOptions, hearingType, hearingSubtype));
+        if (isYes(party.getIsAppointee()) && nonNull(party.getAppointee())) {
+            partyDetails.add(createHearingPartyDetails(party.getAppointee(), hearingOptions, hearingType, hearingSubtype));
+        }
+        if (nonNull(rep) && isYes(rep.getHasRepresentative())) {
+            partyDetails.add(createHearingPartyDetails(rep, hearingOptions, hearingType, hearingSubtype));
+        }
+        return partyDetails;
+    }
+
+    public static PartyDetails createHearingPartyDetails(Entity entity, HearingOptions hearingOptions, String hearingType, HearingSubtype hearingSubtype) {
+        PartyDetailsBuilder partyDetails = PartyDetails.builder();
+
+        partyDetails.partyID(getPartyId(entity));
+        partyDetails.partyType(getPartyType(entity));
+        partyDetails.partyRole(getPartyRole(entity));
+        partyDetails.individualDetails(getPartyIndividualDetails(entity, hearingOptions, hearingType, hearingSubtype));
+        partyDetails.organisationDetails(getPartyOrganisationDetails());
+        partyDetails.unavailabilityDayOfWeek(getPartyUnavailabilityDayOfWeek());
+        partyDetails.unavailabilityRanges(getPartyUnavailabilityRange(hearingOptions));
+
+        return partyDetails.build();
+    }
+
+    public static String getPartyId(Entity entity) {
+        return entity.getId();
+    }
+
+    public static String getPartyType(Entity entity) {
+        return isNotBlank(entity.getOrganisation()) ? ORG.name() : IND.name();
+    }
+
+    public static String getPartyRole(Entity entity) {
```

Comment:
I believe you have this method duplicated

Reply:
Will be fixed when Andrew gets back with the mappings